### DISCUSSION TITLE
worked, run notes

### DISCUSSION THREAD

**Discussion by drhouse** [ottoweiss/pdf-to-audiobook#4] [General]
it worked, cheers :) 
just wanted to leave my notes for anyone else:

`ModuleNotFoundError: No module named 'frontend'`
fixed with: 'pip install PyMuPDF'

`
ModuleNotFoundError: No module named 'tf_keras'
`

> ValueError: Your currently installed version of Keras is Keras 3, but this is not yet supported in Transformers. Please install the backwards-compatible tf-keras package with `pip install tf-keras`.

fixed with: 'pip install tf-keras'
__________________________________________________________________

by any chance, would using GPT-4o cost less or speed anything up?

**Comment by Andy7475** [comment]
even with installing PyMuPDF I got 'module not found frontend'

**Reply by Andy7475** [reply]
I did pip install 'frontend' but got RuntimeError: Directory 'static/' does not exist

**Reply by drhouse** [reply]
here's what I would try, lately I've been using a tool called '[open-interpreter](https://github.com/OpenInterpreter/open-interpreter)', it's AI for your OS:
```
> pip install open-interpreter
> cd (to your pdf-to-audiobook dir)
> interpreter --model openai/gpt-4o-mini --api_key sk-proj-XXXX
> run 'python3 main.py'
```
then see if interpreter can fix the problem, typing 'go' will make it continue if needed

**Reply by Andy7475** [reply]
I got this working by 
pip uninstall fitz
pip install --upgrade --force-reinstall pymupdf

it seems fitz comes packaged with pymupdf, but is different to the version that comes with fitz as a stand-alone package. Before I did this step I also:
mkdir static (as fitz was complaining about the lack of the directory 'static' in the root folder of my project, and also had to pip install 'tools'
However, unsure if you do the first step (uninstalling fitz) whether those steps are then needed or not - but they don't hurt.

Thanks for open-interpreter link @drhouse  - looks like a really useful package
